CI against ruby-head(2.6.0-dev) is getting failed when USE_MINI_PORTILE=true

CI against ruby-head(2.6.0-dev) is getting failed when `USE_MINI_PORTILE=true`
This is due to `sqlite3_libversion_number()` returns `no`.

https://travis-ci.org/sparklemotion/sqlite3-ruby/jobs/380367356#L548

```ruby
checking for sqlite3_libversion_number() in -lsqlite3... no
```

I have attempted to reproduce it locally but it has not reproduced yet so far.

### Attempt to reproduce it

```
rbenv global 2.6.0-dev
git clone https://github.com/sparklemotion/sqlite3-ruby.git
cd sqlite3-ruby
export USE_MINI_PORTILE=true
bundle exec rake test
```

### Environment
- Ubuntu 18.04 LTS (GNU/Linux 4.15.0-20-generic x86_64)
- Installed Ruby version and sqlite3 packages

```
$ ruby -v
ruby 2.6.0dev (2018-05-23 trunk 63494) [x86_64-linux]
$ sudo apt list --installed |grep sqlite
libsqlite3-0/bionic,now 3.22.0-1 amd64 [installed]
libsqlite3-dev/bionic,now 3.22.0-1 amd64 [installed]
sqlite3/bionic,now 3.22.0-1 amd64 [installed]
$
```

Maybe useful info:

### `$ cat tmp/x86_64-linux/sqlite3_native/2.6.0/mkmf.log`


```
"pkg-config --exists sqlite3"
| pkg-config --libs sqlite3
=> "-lsqlite3  \n"
"gcc -o conftest -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/x86_64-linux -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/ruby/backward -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0 -I../../../../ext/sqlite3 -I/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wno-cast-function-type -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-overlength-strings -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wmissing-noreturn  -fPIC conftest.c  -L. -L/home/travis/.rvm/rubies/ruby-head/lib -Wl,-rpath,/home/travis/.rvm/rubies/ruby-head/lib -L/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -Wl,-rpath,/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -L. -fstack-protector -rdynamic -Wl,-export-dynamic     -Wl,-rpath,'/../lib' -Wl,-rpath,'/../lib' -lruby  -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

"gcc -o conftest -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/x86_64-linux -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/ruby/backward -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0 -I../../../../ext/sqlite3 -I/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wno-cast-function-type -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-overlength-strings -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wmissing-noreturn  -fPIC conftest.c  -L. -L/home/travis/.rvm/rubies/ruby-head/lib -Wl,-rpath,/home/travis/.rvm/rubies/ruby-head/lib -L/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -Wl,-rpath,/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -L. -fstack-protector -rdynamic -Wl,-export-dynamic     -Wl,-rpath,'/../lib' -Wl,-rpath,'/../lib' -lruby -lsqlite3 -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

| pkg-config --cflags-only-I sqlite3
=> " \n"
| pkg-config --cflags-only-other sqlite3
=> " \n"
| pkg-config --libs-only-l sqlite3
=> "-lsqlite3  \n"
package configuration for sqlite3
cflags:
ldflags:
libs: -lsqlite3

find_header: checking for sqlite3.h... -------------------- yes

"gcc -E -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/x86_64-linux -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/ruby/backward -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0 -I../../../../ext/sqlite3  -I/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wno-cast-function-type -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-overlength-strings -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wmissing-noreturn  -fPIC   conftest.c -o conftest.i"
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: #include <sqlite3.h>
/* end */

--------------------

find_library: checking for pthread_create() in -lpthread... -------------------- yes

"gcc -o conftest -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/x86_64-linux -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/ruby/backward -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0 -I../../../../ext/sqlite3  -I/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wno-cast-function-type -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-overlength-strings -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wmissing-noreturn  -fPIC  conftest.c  -L. -L/home/travis/.rvm/rubies/ruby-head/lib -Wl,-rpath,/home/travis/.rvm/rubies/ruby-head/lib -L/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -Wl,-rpath,/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -L. -fstack-protector -rdynamic -Wl,-export-dynamic      -lsqlite3 -Wl,-rpath,'/../lib' -Wl,-rpath,'/../lib' -lruby -lpthread  -lsqlite3 -lm   -lc"
conftest.c: In function ‘t’:
conftest.c:14:57: error: ‘pthread_create’ undeclared (first use in this function)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))pthread_create; return !p; }
                                                         ^
conftest.c:14:57: note: each undeclared identifier is reported only once for each function it appears in
conftest.c: At top level:
cc1: warning: unrecognized command line option "-Wno-cast-function-type" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))pthread_create; return !p; }
/* end */

"gcc -o conftest -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/x86_64-linux -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/ruby/backward -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0 -I../../../../ext/sqlite3  -I/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wno-cast-function-type -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-overlength-strings -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wmissing-noreturn  -fPIC  conftest.c  -L. -L/home/travis/.rvm/rubies/ruby-head/lib -Wl,-rpath,/home/travis/.rvm/rubies/ruby-head/lib -L/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -Wl,-rpath,/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -L. -fstack-protector -rdynamic -Wl,-export-dynamic      -lsqlite3 -Wl,-rpath,'/../lib' -Wl,-rpath,'/../lib' -lruby -lpthread  -lsqlite3 -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: extern void pthread_create();
15: int t(void) { pthread_create(); return 0; }
/* end */

--------------------

find_library: checking for sqlite3_libversion_number() in -lsqlite3... -------------------- no

"gcc -o conftest -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/x86_64-linux -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/ruby/backward -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0 -I../../../../ext/sqlite3  -I/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wno-cast-function-type -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-overlength-strings -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wmissing-noreturn  -fPIC  conftest.c  -L. -L/home/travis/.rvm/rubies/ruby-head/lib -Wl,-rpath,/home/travis/.rvm/rubies/ruby-head/lib -L/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -Wl,-rpath,/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -L. -fstack-protector -rdynamic -Wl,-export-dynamic     -lpthread  -lsqlite3 -Wl,-rpath,'/../lib' -Wl,-rpath,'/../lib' -lruby -lsqlite3 -lpthread  -lsqlite3 -lm   -lc"
conftest.c: In function ‘t’:
conftest.c:14:57: error: ‘sqlite3_libversion_number’ undeclared (first use in this function)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_libversion_number; return !p; }
                                                         ^
conftest.c:14:57: note: each undeclared identifier is reported only once for each function it appears in
conftest.c: At top level:
cc1: warning: unrecognized command line option "-Wno-cast-function-type" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_libversion_number; return !p; }
/* end */

"gcc -o conftest -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/x86_64-linux -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0/ruby/backward -I/home/travis/.rvm/rubies/ruby-head/include/ruby-2.6.0 -I../../../../ext/sqlite3  -I/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wno-cast-function-type -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-overlength-strings -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wmissing-noreturn  -fPIC  conftest.c  -L. -L/home/travis/.rvm/rubies/ruby-head/lib -Wl,-rpath,/home/travis/.rvm/rubies/ruby-head/lib -L/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -Wl,-rpath,/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib -L. -fstack-protector -rdynamic -Wl,-export-dynamic     -lpthread  -lsqlite3 -Wl,-rpath,'/../lib' -Wl,-rpath,'/../lib' -lruby -lsqlite3 -lpthread  -lsqlite3 -lm   -lc"
/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.a(sqlite3.o): In function `unixDlError':
sqlite3.c:(.text+0x24921): undefined reference to `dlerror'
/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.a(sqlite3.o): In function `unixDlClose':
sqlite3.c:(.text+0x8244): undefined reference to `dlclose'
/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.a(sqlite3.o): In function `unixDlSym':
sqlite3.c:(.text+0x8257): undefined reference to `dlsym'
/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.a(sqlite3.o): In function `unixDlOpen':
sqlite3.c:(.text+0x8269): undefined reference to `dlopen'
collect2: error: ld returned 1 exit status
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: extern void sqlite3_libversion_number();
15: int t(void) { sqlite3_libversion_number(); return 0; }
/* end */

--------------------
```

### `$ cat tmp/x86_64-pc-linux-gnu/ports/sqlite3/3.8.11.1/compile.log`

```sh
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DPACKAGE_NAME=\"sqlite\" -DPACKAGE_TARNAME=\"sqlite\" -DPACKAGE_VERSION=\"3.8.11.1\" -DPACKAGE_STRING=\"sqlite\ 3.8.11.1\" -DPACKAGE_BUGREPORT=\"http://www.sqlite.org\" -DPACKAGE_URL=\"\" -DPACKAGE=\"sqlite\" -DVERSION=\"3.8.11.1\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_FDATASYNC=1 -DHAVE_USLEEP=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 -DHAVE_DECL_STRERROR_R=1 -DHAVE_STRERROR_R=1 -DHAVE_READLINE=1 -DHAVE_POSIX_FALLOCATE=1 -I.    -D_REENTRANT=1 -DSQLITE_THREADSAFE=1  -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE -O2 -DSQLITE_ENABLE_COLUMN_METADATA -fPIC -MT sqlite3.lo -MD -MP -MF .deps/sqlite3.Tpo -c -o sqlite3.lo sqlite3.c
libtool: compile:  gcc -DPACKAGE_NAME=\"sqlite\" -DPACKAGE_TARNAME=\"sqlite\" -DPACKAGE_VERSION=\"3.8.11.1\" "-DPACKAGE_STRING=\"sqlite 3.8.11.1\"" -DPACKAGE_BUGREPORT=\"http://www.sqlite.org\" -DPACKAGE_URL=\"\" -DPACKAGE=\"sqlite\" -DVERSION=\"3.8.11.1\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_FDATASYNC=1 -DHAVE_USLEEP=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 -DHAVE_DECL_STRERROR_R=1 -DHAVE_STRERROR_R=1 -DHAVE_READLINE=1 -DHAVE_POSIX_FALLOCATE=1 -I. -D_REENTRANT=1 -DSQLITE_THREADSAFE=1 -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE -O2 -DSQLITE_ENABLE_COLUMN_METADATA -fPIC -MT sqlite3.lo -MD -MP -MF .deps/sqlite3.Tpo -c sqlite3.c -o sqlite3.o
mv -f .deps/sqlite3.Tpo .deps/sqlite3.Plo
/bin/bash ./libtool --tag=CC   --mode=link gcc -D_REENTRANT=1 -DSQLITE_THREADSAFE=1  -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE -O2 -DSQLITE_ENABLE_COLUMN_METADATA -fPIC -no-undefined -version-info 8:6:8  -o libsqlite3.la -rpath /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib sqlite3.lo  -ldl -lpthread
libtool: link: ar cru .libs/libsqlite3.a  sqlite3.o
libtool: link: ranlib .libs/libsqlite3.a
libtool: link: ( cd ".libs" && rm -f "libsqlite3.la" && ln -s "../libsqlite3.la" "libsqlite3.la" )
gcc -DPACKAGE_NAME=\"sqlite\" -DPACKAGE_TARNAME=\"sqlite\" -DPACKAGE_VERSION=\"3.8.11.1\" -DPACKAGE_STRING=\"sqlite\ 3.8.11.1\" -DPACKAGE_BUGREPORT=\"http://www.sqlite.org\" -DPACKAGE_URL=\"\" -DPACKAGE=\"sqlite\" -DVERSION=\"3.8.11.1\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_FDATASYNC=1 -DHAVE_USLEEP=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 -DHAVE_DECL_STRERROR_R=1 -DHAVE_STRERROR_R=1 -DHAVE_READLINE=1 -DHAVE_POSIX_FALLOCATE=1 -I.    -D_REENTRANT=1 -DSQLITE_THREADSAFE=1  -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE -O2 -DSQLITE_ENABLE_COLUMN_METADATA -fPIC -MT sqlite3-shell.o -MD -MP -MF .deps/sqlite3-shell.Tpo -c -o sqlite3-shell.o `test -f 'shell.c' || echo './'`shell.c
mv -f .deps/sqlite3-shell.Tpo .deps/sqlite3-shell.Po
gcc -DPACKAGE_NAME=\"sqlite\" -DPACKAGE_TARNAME=\"sqlite\" -DPACKAGE_VERSION=\"3.8.11.1\" -DPACKAGE_STRING=\"sqlite\ 3.8.11.1\" -DPACKAGE_BUGREPORT=\"http://www.sqlite.org\" -DPACKAGE_URL=\"\" -DPACKAGE=\"sqlite\" -DVERSION=\"3.8.11.1\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_FDATASYNC=1 -DHAVE_USLEEP=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 -DHAVE_DECL_STRERROR_R=1 -DHAVE_STRERROR_R=1 -DHAVE_READLINE=1 -DHAVE_POSIX_FALLOCATE=1 -I.    -D_REENTRANT=1 -DSQLITE_THREADSAFE=1  -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE -O2 -DSQLITE_ENABLE_COLUMN_METADATA -fPIC -MT sqlite3-sqlite3.o -MD -MP -MF .deps/sqlite3-sqlite3.Tpo -c -o sqlite3-sqlite3.o `test -f 'sqlite3.c' || echo './'`sqlite3.c
mv -f .deps/sqlite3-sqlite3.Tpo .deps/sqlite3-sqlite3.Po
/bin/bash ./libtool --tag=CC   --mode=link gcc -D_REENTRANT=1 -DSQLITE_THREADSAFE=1  -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE -O2 -DSQLITE_ENABLE_COLUMN_METADATA -fPIC   -o sqlite3 sqlite3-shell.o sqlite3-sqlite3.o -lreadline -lcurses  -ldl -lpthread
libtool: link: gcc -D_REENTRANT=1 -DSQLITE_THREADSAFE=1 -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE -O2 -DSQLITE_ENABLE_COLUMN_METADATA -fPIC -o sqlite3 sqlite3-shell.o sqlite3-sqlite3.o  -lreadline -lcurses -ldl -lpthread
```

### `$ cat tmp/x86_64-pc-linux-gnu/ports/sqlite3/3.8.11.1/configure.log`

```sh
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for x86_64-pc-linux-gnu-strip... no
checking for strip... strip
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking for x86_64-pc-linux-gnu-gcc... no
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... gcc3
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for x86_64-pc-linux-gnu-gcc... gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking dependency style of gcc... (cached) gcc3
checking for x86_64-pc-linux-gnu-ranlib... no
checking for ranlib... ranlib
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... no
checking for x86_64-pc-linux-gnu-dumpbin... no
checking for x86_64-pc-linux-gnu-link... no
checking for dumpbin... no
checking for link... link -dump
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for x86_64-pc-linux-gnu-objdump... no
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for x86_64-pc-linux-gnu-dlltool... no
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for x86_64-pc-linux-gnu-ar... no
checking for ar... ar
checking for archiver @FILE support... @
checking for x86_64-pc-linux-gnu-strip... strip
checking for x86_64-pc-linux-gnu-ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for x86_64-pc-linux-gnu-mt... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for fdatasync... yes
checking for usleep... yes
checking for fullfsync... no
checking for localtime_r... yes
checking for gmtime_r... yes
checking whether strerror_r is declared... yes
checking for strerror_r... yes
checking whether strerror_r returns char *... no
checking for library containing tgetent... -lcurses
checking for library containing readline... -lreadline
checking for readline... yes
checking for library containing pthread_create... -lpthread
checking for library containing dlopen... -ldl
checking for whether to support dynamic extensions... yes
checking for posix_fallocate... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating sqlite3.pc
config.status: executing depfiles commands
config.status: executing libtool commands
```

### `$ cat tmp/x86_64-pc-linux-gnu/ports/sqlite3/3.8.11.1/configure.md5`

```sh
a792244ca55125e37018f8b541d2245e
```

### `$ cat tmp/x86_64-pc-linux-gnu/ports/sqlite3/3.8.11.1/install.log`

```sh
make[1]: Entering directory `/home/travis/build/JoshCheek/sqlite3-ruby/tmp/x86_64-pc-linux-gnu/ports/sqlite3/3.8.11.1/sqlite-autoconf-3081101'
 /bin/mkdir -p '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib'
 /bin/bash ./libtool   --mode=install /usr/bin/install -c   libsqlite3.la '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib'
libtool: install: /usr/bin/install -c .libs/libsqlite3.lai /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.la
libtool: install: /usr/bin/install -c .libs/libsqlite3.a /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.a
libtool: install: chmod 644 /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.a
libtool: install: ranlib /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/libsqlite3.a
libtool: finish: PATH="/home/travis/build/JoshCheek/sqlite3-ruby/vendor/bundle/ruby/2.6.0/bin:/home/travis/.local/bin:/opt/pyenv/shims:/home/travis/.phpenv/shims:/home/travis/perl5/perlbrew/bin:/home/travis/.kiex/elixirs/elixir-1.4.5/bin:/home/travis/.kiex/bin:/home/travis/gopath/bin:/home/travis/.gimme/versions/go1.7.4.linux.amd64/bin:/usr/local/phantomjs/bin:/usr/local/phantomjs:/usr/local/neo4j-3.2.7/bin:/usr/local/maven-3.5.2/bin:/usr/local/cmake-3.9.2/bin:/usr/local/clang-5.0.0/bin:/home/travis/.rvm/gems/ruby-head/bin:/home/travis/.rvm/gems/ruby-head@global/bin:/home/travis/.rvm/rubies/ruby-head/bin:/home/travis/.rvm/bin:/home/travis/bin:/home/travis/.nvm/versions/node/v8.9.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/travis/.phpenv/bin:/opt/pyenv/bin:/home/travis/.yarn/bin:/home/travis/.debug:/sbin" ldconfig -n /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib
----------------------------------------------------------------------
Libraries have been installed in:
   /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------
 /bin/mkdir -p '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/bin'
  /bin/bash ./libtool   --mode=install /usr/bin/install -c sqlite3 '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/bin'
libtool: install: /usr/bin/install -c sqlite3 /home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/bin/sqlite3
 /bin/mkdir -p '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include'
 /usr/bin/install -c -m 644 sqlite3.h sqlite3ext.h '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/include'
 /bin/mkdir -p '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/share/man/man1'
 /usr/bin/install -c -m 644 sqlite3.1 '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/share/man/man1'
 /bin/mkdir -p '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/pkgconfig'
 /usr/bin/install -c -m 644 sqlite3.pc '/home/travis/build/JoshCheek/sqlite3-ruby/ports/x86_64-pc-linux-gnu/sqlite3/3.8.11.1/lib/pkgconfig'
make[1]: Leaving directory `/home/travis/build/JoshCheek/sqlite3-ruby/tmp/x86_64-pc-linux-gnu/ports/sqlite3/3.8.11.1/sqlite-autoconf-3081101'
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,6 +12,9 @@ addons:
 after_failure:
   - "find . -name mkmf.log -exec cat {} \\;"
 
+after_success:
+  - "find . -name mkmf.log -exec cat {} \\;"
+
 env:
   - USE_MINI_PORTILE=true
   - USE_MINI_PORTILE=false

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,9 @@ addons:
     packages:
       - libgmp-dev
 
+after_failure:
+  - "find . -name mkmf.log -exec cat {} \\;"
+
 env:
   - USE_MINI_PORTILE=true
   - USE_MINI_PORTILE=false

--- a/ext/sqlite3/extconf.rb
+++ b/ext/sqlite3/extconf.rb
@@ -64,6 +64,12 @@ end
 asplode('sqlite3.h')  unless find_header  'sqlite3.h'
 find_library 'pthread', 'pthread_create' # 1.8 support. *shrug*
 
+have_library 'dl'
+
+%w{ dlopen dlclose dlsym }.each do |func|
+  abort "missing function #{func}" unless have_func(func)
+end
+
 if with_config('sqlcipher')
   asplode('sqlcipher') unless find_library 'sqlcipher', 'sqlite3_libversion_number'
 else

